### ISSUE TITLE
browserWindow 'closed' event handler not getting invoked.

### ISSUE THREAD

**Issue by ssreekanth** [Quramy/electron-connect#7]
I have the following sample code along with other .html files for main window and child window. You can find the entire code at the repo url - https://bitbucket.org/ssreekanth/sample-electron-connect.git

```
var app = require('app'),
      BrowserWindow = require('browser-window'),
      ipc = require('ipc');

  var mainWindow = null;
  var childWindow = null;

  app.on('window-all-closed', function() {
      if (process.platform != 'darwin') {
          app.quit();
      }
  });

  app.on('ready', function() {
      mainWindow = new BrowserWindow({
          width: 480,
          height: 640,
          resizable: false
      });

      mainWindow.loadUrl('file://' + __dirname + '/main.html');

      mainWindow.on('closed', function() {
          mainWindow = null;
      });

      ipc.on('child:open', function(event) {
          console.log('main.js :: ipc msg child:open received');
          openChildWindow();
      });

      openChildWindow = function() {
          childWindow = new BrowserWindow({
              width: 1080,
              height: 640,
          });

          childWindow.loadUrl('file://' + __dirname + '/child.html');

          childWindow.on('closed', function() {
              console.log('child window closed');
              childWindow = null;
          });
      };
  });
```

When not integrated with 'electron-connect', all works fine. childWindow.on('closed', function() {}) gets invoked and I could see the 'child window closed' message getting printed on the console.

However, when I enable electron-connect, I don't see this handler getting invoked. Looks like electron-connect is handling the 'close' event, but the app doesn't see the same event.

What could be wrong? How to fix this?

**Comment by ssreekanth** [comment]
@Quramy any idea/suggestion to fix this?

**Comment by Quramy** [comment]
@ssreekanth 

I reproduced your issue. 

electron-connect unregister all listeners when `beforeunload`.
So your `close` handler isn't invoked.

It's a electron-connect's bug and I will fix this.

**Comment by ssreekanth** [comment]
@Quramy :+1: I'll look forward for the fix.

**Comment by Quramy** [comment]
Hi, @ssreekanth .

I fixed this issue at v0.3.1 . Please try to check it out.

**Comment by ssreekanth** [comment]
Thanks @Quramy for the quick fix. The sample code works fine, i'm closing this issue.
I'll test it out with my real code, and let you know if I see any  issues there.
